src/api/preInstallApi.ts

```diff
@@ -18,17 +18,18 @@ export const getFields = async (
         )}`,
       },
     }
-  ).then((res) => res.json());
+  );
+
+  return res.json() as Promise<FieldMeta[]>;
```

Comment:
There is no validation to prove that this is of type `FieldMeta`. We should check the response code and optionally the body format.